plugin/src/main/groovy/com/novoda/staticanalysis/internal/findbugs/FindbugsConfigurator.groovy

```diff
@@ -118,17 +120,27 @@ class FindbugsConfigurator extends CodeQualityConfigurator<FindBugs, FindBugsExt
     }
 
     private FileCollection getJavaClasses(SourceSet sourceSet, List<String> includes) {
-        includes.isEmpty() ? project.files() : createClassesTreeFrom(sourceSet).include(includes)
+        includes.isEmpty() ? project.files() : createClassesTreeFrom(sourceSet, includes)
     }
 
     /**
      * The simple "classes = sourceSet.output" may lead to non-existing resources directory
      * being passed to FindBugs Ant task, resulting in an error
      * */
-    private ConfigurableFileTree createClassesTreeFrom(SourceSet sourceSet) {
-        project.fileTree(sourceSet.output.classesDir, {
-            it.builtBy(sourceSet.output)
-        })
+    private FileTree createClassesTreeFrom(SourceSet sourceSet, List<String> includes) {
+        def fileTrees = sourceSet.output.classesDirs.collect { classesDir ->
+            project.fileTree(classesDir) {
+                builtBy(sourceSet.output)
+                include(includes)
+            }
+        }
+        if (fileTrees.isEmpty()) {
+            throw GradleException("No file tree from ${sourceSet} and filters: ${includes}")
+        } else if (fileTrees.size() == 1) {
+            fileTrees.first()
+        } else {
+            fileTrees.inject(fileTrees.head()) { result, current -> result + current }
```

Comment:
I believe we should be able to `inject` while iterating on `classesDirs`, this code could be a lot simpler in the end, something like [this](https://github.com/novoda/gradle-static-analysis-plugin/blob/master/plugin/src/main/groovy/com/novoda/staticanalysis/internal/SourceFilter.groovy#L40):
```
classesDirs.inject(null) { tree, file -> tree?.plus(project.fileTree(file)) ?: project.fileTree(file) }.include(includes)
```